src/pages/content-edit/head.js

```diff
@@ -3,261 +3,431 @@ import format from "date-fns/format";
 import isFuture from "date-fns/is_future";
 import isPast from "date-fns/is_past";
 import subSeconds from "date-fns/sub_seconds";
-import compareDesc from "date-fns/compare_desc";
-import get from "lodash.get";
+import clone from "lodash.clone";
 import upper from "lodash.capitalize";
+import get from "lodash.get";
 
 import config, { icons } from "../../config";
 import db from "../../lib/firebase";
 
 import css from "./head.css";
 
+var DEFAULT_START_TIME = "00:00",
+    DEFAULT_END_TIME   = "23:59",
+    DATE_FORMAT = "YYYY-MM-DD",
+    TIME_FORMAT = "HH:mm";
+
 export function controller(options) {
     var ctrl = this,
         ref  = options.ref,
         user = db.getAuth().uid,
 
-        publish   = options.data.published_at ? options.data.published_at : null,
-        unpublish = options.data.unpublished_at ? options.data.unpublished_at : null;
+        publishTs   = options.data.published_at ? options.data.published_at : null,
+        unpublishTs = options.data.unpublished_at ? options.data.unpublished_at : null,
 
-    ctrl.schedule   = false;
-    ctrl.unschedule = false;
+        defaultStartTime = get(config, "defaults.publish_start_time") || DEFAULT_START_TIME,
+        defaultEndTime   = get(config, "defaults.publish_end_time")   || DEFAULT_END_TIME;
 
-    if(publish) {
-        ctrl.start = {
-            date : format(publish, "YYYY-MM-DD"),
-            time : format(publish, "HH:mm")
-        };
-    } else {
-        ctrl.start = {
-            date : "",
-            time : get(config, "defaults.publish_start_time") || ""
-        };
-    }
+    ctrl.init = function() {
+        ctrl.schedule   = false;
 
-    if(unpublish) {
-        ctrl.end = {
-            date : format(unpublish, "YYYY-MM-DD"),
-            time : format(unpublish, "HH:mm")
-        };
-    } else {
-        ctrl.end = {
-            date : "",
-            time : ""
-        };
-    }
+        if(publishTs) {
```

Comment:
I have this knee jerk reaction against using ternaries if the line gets longer than a certain length.
```
ctrl.start = (publishTs) ? ctrl.makeScheduleObj(publishTs) : ctrl.nulledDate(defaultStartTime);
```
at its current indentation goes beyond my `100` character ruler in Sublime. I'll change these though.